bundle/lib/source/DobbyConfig.cpp

```diff
@@ -505,6 +505,32 @@ void DobbyConfig::addPluginLauncherHooks(std::shared_ptr<rt_dobby_schema> cfg, c
     cfg->hooks->start_container[cfg->hooks->start_container_len-1] = startContainerEntry;
 }
 
+// -----------------------------------------------------------------------------
+/**
+ *  @brief Sets the container hostname to the container ID
+ *
+ *  @param[in]  id              container identifier
+ *  @param[in]  cfg             libocispec config structure instance
+ *  @param[in]  bundlePath      path to the container bundle
+ */
+bool DobbyConfig::setHostnameToContainerId(const ContainerId& id, std::shared_ptr<rt_dobby_schema> cfg, const std::string& bundlePath)
+{
+    // change hostname to container id only if necessary
+    if (!strcmp(cfg->hostname, id.c_str()))
+    {
+        return true;
+    }
+    cfg->hostname = strdup(id.c_str());
```

Comment:
We should use free before using strdup. Using strdup allocates new memory, meaning old pointer is lost before we used free on it. So we have memory leak here. Strange enough I have found some other uses of strdup without free in our code. We should fix those also.